src/sqlfluff/dialects/dialect_spark3.py

```diff
@@ -794,6 +794,53 @@ class UseDatabaseStatementSegment(BaseSegment):
     )
 
 
+# Data Manipulation Statements
+@spark3_dialect.segment(replace=True)
+class InsertStatementSegment(BaseSegment):
+    """A `INSERT [TABLE]` statement to insert or overwrite new rows into a table.
+
+    https://spark.apache.org/docs/latest/sql-ref-syntax-dml-insert-into.html
+    https://spark.apache.org/docs/latest/sql-ref-syntax-dml-insert-overwrite-table.html
+    """
+
+    type = "insert_table_statement"
+
+    match_grammar = Sequence(
+        "INSERT",
+        OneOf("INTO", "OVERWRITE"),
+        Ref.keyword("TABLE", optional=True),
+        Ref("TableReferenceSegment"),
+        Ref("PartitionSpecGrammar", optional=True),
+        Ref("BracketedColumnReferenceListGrammar", optional=True),
+        OneOf(
+            AnyNumberOf(
+                Ref("ValuesClauseSegment"),
+                min_times=1,
+            ),
+            Ref("SelectableGrammar"),
+            Sequence(
+                Ref.keyword("TABLE", optional=True),
+                Ref("TableReferenceSegment"),
+            ),
+            Sequence(
+                # TODO is there a less brittle way to do this using an existing class(es)?
+                # TODO if not, how to best construct a new class for `SELECT` match?
```

Comment:
Let's remove these, as can't honestly think of a better way to do this, other than splitting up Select Segment hugely which seems overkill for this one usecase.

```suggestion
```

Reply:
👍